### PR TITLE
ssl: first pass limit when allocating buffer for certificates

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5188

Describe changes:
- Adds a first time check before allocating much memory to store a certificate

Feedback welcome here

### CODE REVIEW

File: src/app-layer-ssl.c [hunk 1]
```diff
@@ -1431,6 +1431,10 @@ static int EnsureRecordSpace(SSLStateConnp *curr_connp, const uint8_t * const in
         SCLogDebug("cert_len unknown still, create small buffer to start");
         certs_len = 256;
     }
+    // Limit in a first time allocation for very large certificates
+    if (certs_len > 0x10000 && certs_len > curr_connp->trec_pos + input_len) {
```
Comment (victorjulien) [inline src/app-layer-ssl.c:1435]: should we first check if current input will set the limited value?

Also please use a macro with a clear name, a decimal value and a comment to explain the default

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 6537

File: src/app-layer-ssl.c (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-ssl.c:1435]: > should we first check if current input will set the limited value?

I am not sure I understand your question.
This is a first check that current input will set the limited value...

That is if not `certs_len > curr_connp->trec_pos + input_len` that means that the current input will cross the 65k boundary, and thus, we do not limit ourselves

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7139

**Pull request closed without merging**